src/gui/elevation/qgselevationcontrollerwidget.cpp

```diff
@@ -0,0 +1,290 @@
+  // on the widget itself to be ignored and passed to underlying widgets which are NOT THE DIRECT
+  // PARENT of this widget.
+  // this is definitively *****NOT***** possible with event filters, by overriding mouse events, or
+  // with the WA_TransparentForMouseEvents attribute
+
+  QRegion reg( frameGeometry() );
+  reg -= QRegion( geometry() );
+  reg += childrenRegion();
+  setMask( reg );
+}
+
+//
+// QgsElevationControllerLabels
+//
+///@cond PRIVATE
+QgsElevationControllerLabels::QgsElevationControllerLabels( QWidget *parent )
+  : QWidget( parent )
+{
+  const QFontMetrics fm( font() );
+  setMinimumWidth( fm.horizontalAdvance( '0' ) * 5 );
+  setAttribute( Qt::WA_TransparentForMouseEvents );
+}
+
+void QgsElevationControllerLabels::paintEvent( QPaintEvent * )
+{
+  QStyleOptionSlider styleOption;
+  styleOption.initFrom( this );
+
+  const QRect sliderRect = style()->subControlRect( QStyle::CC_Slider, &styleOption,  QStyle::SC_SliderHandle, this );
+  const int sliderHeight = sliderRect.height();
+
+  QFont f = font();
+  const QFontMetrics fm( f );
+
+  const int left = rect().left() + 2;
+
+  const double limitRange = mLimits.upper() - mLimits.lower();
+  const double lowerFraction = ( mRange.lower() - mLimits.lower() ) / limitRange;
+  const double upperFraction = ( mRange.upper() - mLimits.lower() ) / limitRange;
+  const int lowerY = static_cast< int >( std::round( rect().bottom() - sliderHeight * 0.5 - ( rect().height() - sliderHeight ) * lowerFraction - fm.descent() ) );
+  const int upperY = static_cast< int >( std::round( rect().bottom() - sliderHeight * 0.5 - ( rect().height() - sliderHeight ) * upperFraction + fm.ascent() ) );
+
+  const bool lowerIsCloseToLimit = lowerY + fm.height() > rect().bottom() - fm.descent();
+  const bool upperIsCloseToLimit = upperY - fm.height() < rect().top() + fm.ascent();
+
+  QLocale locale;
+
+  QPainterPath path;
+  if ( mLimits.lower() > std::numeric_limits< double >::lowest() )
+  {
+    if ( lowerIsCloseToLimit )
+    {
+      f.setBold( true );
+      path.addText( left, lowerY, f, locale.toString( mRange.lower() ) );
+    }
+    else
+    {
+      f.setBold( true );
+      path.addText( left, lowerY, f, locale.toString( mRange.lower() ) );
+      f.setBold( false );
+      path.addText( left, rect().bottom() - fm.descent(), f, locale.toString( mLimits.lower() ) );
+    }
+  }
+
+  if ( mLimits.upper() < std::numeric_limits< double >::max() )
+  {
+    if ( upperIsCloseToLimit )
+    {
+      f.setBold( true );
+      path.addText( left, upperY, f, locale.toString( mRange.upper() ) );
+    }
+    else
+    {
+      f.setBold( true );
+      path.addText( left, upperY, f, locale.toString( mRange.upper() ) );
+      f.setBold( false );
+      path.addText( left, rect().top() + fm.ascent(), f, locale.toString( mLimits.upper() ) );
+    }
+  }
+
+  QPainter p( this );
+  p.setRenderHint( QPainter::Antialiasing, true );
+  const QColor bufferColor = palette().color( QPalette::Window );
```

Comment:
Thanks for not using hard-coded values here, this'll play nicely with map themes.